packages/video_player/ios/Classes/VideoPlayerPlugin.m

```diff
@@ -155,8 +161,18 @@ - (void)createVideoOutputAndDisplayLink:(FLTFrameUpdater*)frameUpdater {
   _displayLink.paused = YES;
 }
 
-- (instancetype)initWithURL:(NSURL*)url frameUpdater:(FLTFrameUpdater*)frameUpdater {
-  AVPlayerItem* item = [AVPlayerItem playerItemWithURL:url];
+- (instancetype)initWithURL:(NSURL*)url
+               frameUpdater:(FLTFrameUpdater*)frameUpdater
+                   mimeType:(NSString*)mimeType {
+  NSString* fileExtention = [[url absoluteString] pathExtension];
+  AVURLAsset* asset;
+  if ([fileExtention length] == 0) {
+    asset = [[AVURLAsset alloc] initWithURL:url
+                                    options:@{@"AVURLAssetOutOfBandMIMETypeKey" : mimeType}];
```

Comment:
`AVURLAssetOutOfBandMIMETypeKey` isn't publicly listed as part of the option keys ([documentation](https://developer.apple.com/documentation/avfoundation/avurlasset/initialization_options?language=objc)).
How can we tell if it's safe to use?

Reply:
You got me.I haven't found another way to specify mime type of `AVAsset`. Maybe we can obfuscate it if that is acceptable, if not I'll just close this PR.
Do you know any other workaround about this issue, since the url comes from a third-party SDK that i cannot modify.If there's no other way I can only take the risk using this solution.

Reply:
I do not have a concrete way to work around this.
An unverified idea: maybe you can use NSURLProtocol to add a content type to the http response?

Anyway I don't think we should land this as-is, let's close this PR, at least for now.

Reply:
OK